R/utilities-mpi-error-handling.R

```diff
@@ -0,0 +1,153 @@
+#' @title checkSimulationLoaded
+#' @description Check that the simulation has been loaded on the core.
+#' @param simulation the simulation object to be checked if loaded successfully
+#' @param simulationFilePath the path to the simulation object to be checked if loaded successfully
+#' @export
+checkSimulationLoaded <- function(simulation, simulationFilePath) {
+  success <- FALSE
+  if (identical(simulation$sourceFile, simulationFilePath)) {
+    success <- TRUE
+  }
+  return(success)
+}
+
+#' @title checkLibraryLoaded
+#' @description Check that a library has been successfully loaded on the core.
+#' @param libraryName the library to be checked if loaded successfully
+#' @export
+checkLibraryLoaded <- function(libraryName) {
+  success <- FALSE
+  if (libraryName %in% .packages()) {
+    success <- TRUE
+  }
+  return(success)
+}
+
+
+#' @title checkNotNull
+#' @description Check that an object is not null.
+#' @param object to be checked if not null
+#' @export
+checkNotNull <- function(object) {
+  success <- FALSE
+  if (!is.null(object)) {
+    success <- TRUE
+  }
+  return(success)
+}
+
+#' @title checkAllCoresSuccessful
+#' @description Check that all cores executed an mpi.remote.exec command successfully.
+#' @param coreResults list of results returned by each core after an mpi.remote.exec command is complete
+checkAllCoresSuccessful <- function(coreResults) {
+  allCoreResults <- sapply(1:length(coreResults), function(n, obj) obj[[n]], coreResults)
+  return(all(allCoreResults))
+}
+
+#' @title loadSimulationOnCores
+#' @description Send structureSet to core, check its simulation has been loaded successfully
+#' @param structureSet containing simulationSet which contains path to simulation file and pathIDs to be loaded in simulation object as outputs
+loadSimulationOnCores <- function(structureSet, logFolder = getwd()) {
```

Comment:
I don't think we should have default value for logFolder in utilities method. This is a mandatory parameter

Reply:
Removed